docs/proposals/ns-scoped-catalog.md

```diff
@@ -0,0 +1,256 @@
+
+- `ServicePlan.Spec` should have `ServiceBrokerName` instead of `ClusterServiceBrokerName`
+- `ServicePlan.Spec` should have `ServiceClassRef` instead of `ClusterServiceClassRef`
+
+### Changes to `ServiceInstance`
+
+The `ServiceInstance` resource should be changed to allow users to unambiguously specify a `ServiceClass` and `ServicePlan` instead of the cluster-level resources.
+
+- Add fields to `PlanReference` to represent the external and k8s names of `ServiceClass` and `ServicePlan` (as opposed to the cluster-scoped versions)
+- Add reference fields to `ServiceInstanceSpec` that represent the references to the namespaced resources
+
+## Implementation plan
+
+The implementation of this proposal is too large for a single PR, so we'll break it into stages:
+
+### Extracting shared fields from existing resources
+
+To extract the shared fields, we will:
+
+- Extract the identified shared fields onto embedable types: `SharedServiceBrokerSpec`, `SharedServiceClassSpec`, and `SharedServicePlanSpec`.
+- Embed these shared types within their respective cluster scoped specs.
+- Minor controller changes to reflect the fact that these fields now belong to an embedded type.
+- Required updates to the fuzzer, validations, and defaults where necessary.
+
+This PR will result in no behavioral changes, and should remain entirely transparent to users. It is effectively a no-op.
+
+Relevant API types after this step:
+
+```go
+
+type ClusterServiceBrokerSpec struct {
+    SharedServiceBrokerSpec `json:",inline"`
+    
+    AuthInfo *ClusterServiceBrokerAuthInfo `json:"authInfo,omitempty"`
+}
+
+type ClusterServiceBrokerStatus struct {
+    SharedServiceBrokerStatus `json:",inline"`
+}
+
+type ClusterServiceClassSpec struct {
+    SharedServiceClassSpec `json:",inline"`
+    
+    ClusterServiceBrokerName string `json:"clusterServiceBrokerName"`
+}
+
+type ClusterServiceClassStatus struct {
+    SharedServiceClassStatus `json:",inline"`
+}
+
+type ClusterServicePlanSpec struct {
+    SharedServicePlanSpec `json:",inline"`
+    
+    ClusterServiceBrokerName string `json:"clusterServiceBrokerName"`
+    
+    ClusterServiceClassRef ClusterObjectReference `json:"clusterServiceClassRef"`
+}
+
+type ClusterServicePlanStatus struct {
+    SharedServicePlanStatus `json:",inline"`
+}
+```
+
+### Add API surface for ns-scoped resources
+
+After shared fields are extracted, we will add:
+
+- API resources for the new namespace-scoped resources
+- Associated validations / fuzzers / client changes / etc
+- A feature gate that controls whether the ns-scoped resources are enabled
+
+The feature gate is necessary to ensure that users do not see resources that aren't fully functional and will be removed after the proposal is completely implemented.
+
+The new resources will look as follows:
+
+```go
+type ServiceBrokerSpec struct {
+    SharedServiceBrokerSpec `json:",inline"`
+    
+    AuthInfo *ServiceBrokerAuthInfo `json:"authInfo,omitempty"`
```

Comment:
I think this is what I mean by `localref` version of struct above.

Reply:
This is the old name, but the content needs to switch to some kind of `localref`.

Reply:
Yeah, that's what's implied. I can add the type description here if people think it's necessary